sphinx_gallery/gen_rst.py

```diff
@@ -626,20 +642,23 @@ def execute_script(script_blocks, script_vars, gallery_conf):
     time_elapsed : float
         Time elapsed during execution
     """
-
-    example_globals = {
+    # Examples may contain if __name__ == '__main__' guards
+    # for in example scikit-learn if the example uses multiprocessing.
+    # Here we create a new __main__ module, and temporarily change
+    # sys.modules when running our example
+    fake_main = imp.new_module("__main__")
```

Comment:
Should we use `importlib.util.module_from_spec()` instead as `imp` is deprecated? https://docs.python.org/3/library/imp.html#imp.new_module